include/aie/Dialect/AIE/IR/AIE.td

```diff
@@ -1024,30 +1025,32 @@ def AIE_LockOp: AIE_Op<"lock", [TileElement]>, Results<(outs Index)> {
     TileOp getTileOp();
   }];
   let builders = [
-    OpBuilder<(ins "Value":$tile, "int":$lockID), [{
+    OpBuilder<(ins "Value":$tile, "int":$lockID, "int":$init), [{
       build($_builder, $_state,
         $_builder.getIndexType(),
         tile,
-        $_builder.getI32IntegerAttr(lockID));
+        $_builder.getI32IntegerAttr(lockID),
+        $_builder.getI32IntegerAttr(init));
       }]>
   ];
   let hasVerifier = 1;
   // let hasCanonicalizer = 1;
 }
 
-
 def AIE_UseLockOp: AIE_Op<"useLock", []> {
   let summary = "acquire/release lock op";
   let description = [{
-    This operation uses a lock. A lock can be acquired with a value, or release with a value.
-    This should be understood as a "blocking" operation.  This lock must appear in a parent op
-    where the tile can be determined (A CoreOp, a ShimDMAOp, or a MemOp).  If the useLock
-    operation appears in a module directly, an initialization to the lock will be generated in
-    the host implementation.
+    This operation uses a lock. In AIE1, a lock can be acquired with a value,
+    or released with a value. This should be understood as a "blocking"
+    operation. In AIE2, locks are counting semaphores without inherent
+    acquired/release characteristic. This lock must appear in a parent op where
+    the tile can be determined (A CoreOp, a ShimDMAOp, or a MemOp). If the
+    useLock operation appears in a module directly, an initialization to the
```

Comment:
If we're moving to having an init value, then I'd like to see the old usage of useLock removed.

Reply:
This is addressed in [14ae4f3](https://github.com/Xilinx/mlir-aie/pull/457/commits/14ae4f3ede310ef18b7dcc2ea39e70ded487615a) but I haven't fully tested it.